src/library/AsyncWithdrawLib.sol

```diff
@@ -23,43 +25,57 @@ library AsyncWithdrawLib {
         uint256 assetListLength = assetList.length;
 
         for (uint256 i = 0; i < assetListLength; i++) {
-            (, uint256 balanceInBase) = asyncWithdrawBalance(assetList[i]);
+            (, uint256 balanceInBase) = asyncWithdrawBalance(assetList[i], address(this));
             totalBaseBalance += balanceInBase;
         }
     }
 
     /**
-     * @notice Internal function to handle the assets that are in queue for withdrawal.
+     * @notice function to handle the assets that are in queue for withdrawal.
      * @param asset_ The address of the asset.
      * @dev This function should return the amount in base denomination.
      */
-    function asyncWithdrawBalance(address asset_) public view returns (uint256 assets, uint256 baseAssets) {
-        assets = getAssets(asset_);
+    function asyncWithdrawBalance(address asset_, address owner)
+        public
+        view
+        returns (uint256 assets, uint256 baseAssets)
+    {
+        assets = getQueuedAssets(asset_, owner);
         baseAssets = VaultLib.convertAssetToBase(asset_, assets);
     }
 
-    function getAssets(address asset) public pure returns (uint256 assets) {
-        if (asset == MC.WETH) {
-            return 0;
+    function getAssetsQueuedForWithdrawal(address queueManager_, address owner) public view returns (uint256 assets) {
+        IWithdrawalQueueManager queueManager = IWithdrawalQueueManager(queueManager_);
+        (, IWithdrawalQueueManager.WithdrawalRequest[] memory requests) = queueManager.withdrawalRequestsForOwner(owner);
+        for (uint256 i = 0; i < requests.length; i++) {
+            if (!requests[i].processed) {
+                assets += requests[i].amount;
+            }
         }
+        return assets;
+    }
 
-        if (asset == MC.BUFFER) {
-            return 0;
-        }
+    function getQueuedAssets(address asset, address owner) public view returns (uint256 assets) {
+        // TODO: support WOETH
 
-        if (asset == MC.STETH) {
-            return 0;
+        if (asset == MC.WSTETH) {
+            IWithdrawalQueue queue = IWithdrawalQueue(MC.WSTETH_WQ);
+            uint256[] memory requestIds = queue.getWithdrawalRequests(owner);
+            IWithdrawalQueue.WithdrawalRequestStatus[] memory statuses = queue.getWithdrawalStatus(requestIds);
+            for (uint256 i = 0; i < statuses.length; i++) {
+                assets += statuses[i].amountOfShares;
```

Comment:
i haven't looked into this in depth, i am curious, while you're in a withdrawal state, do you still earn yield with WSTETH if this is priced in shares then?

because for ynETH and ynLSDe as mentioned below we stop giving you yield

Reply:
When claiming the share rate is calculated here:

https://github.com/lidofinance/core/blob/901c0e19b752f5ea03118d71881d8b72ccee44c0/contracts/0.8.9/WithdrawalQueueBase.sol#L507

I'm not sure how this maxShareRate is calculated, but it is input from offchain through an oracle to this queue when finalizing a checkpoint.

Reply:
It looks like it doesn't really matter the share rate, since they also provide amountOfStETH which is basically the underlying assets in base which is what we require. So we don't need to do any extra calculations here.